### ISSUE TITLE
Train With My Own Dataset

### ISSUE THREAD

**Issue by Angzz** [sfzhang15/RefineDet#61]
Hi, I think your idea is great, and I want to use your method to deal with my own dataset, so what should I change if do like that? 

Thanks!!!

**Comment by sfzhang15** [comment]
@Angzz 
Hi, you just need modify the number class according to your dataset, the code will be run successfully. But if you want to get better results, you should modify the anchor (aspect ratio, scale, layer), data augmentation and other parameters, e.g., IoU threshold to divide positives and negatives in ARM and ODM.

**Comment by Angzz** [comment]
@sfzhang15 Hi, I have successfully generate the lmdb file with my own dataset, and I changed somthing else like num_class to filt my own dataset, but when I train the model, I find all the loss is 0, can you give me some suggestions? 

Thanks!!!

Part of the train log is shown below, I change the resize_with and resize_height to 640, but when I recover to 512, it did't work either. 

I0524 15:49:23.859304 15114 solver.cpp:259]     Train net output #0: arm_loss = 0 (* 1 = 0 loss)
I0524 15:49:23.859360 15114 solver.cpp:259]     Train net output #1: odm_loss = 0 (* 1 = 0 loss)
I0524 15:49:23.859488 15114 sgd_solver.cpp:138] Iteration 30, lr = 0.001
I0524 15:49:32.134764 15114 solver.cpp:243] Iteration 40, loss = 0
I0524 15:49:32.136090 15114 solver.cpp:259]     Train net output #0: arm_loss = 0 (* 1 = 0 loss)
I0524 15:49:32.136328 15114 solver.cpp:259]     Train net output #1: odm_loss = 0 (* 1 = 0 loss)
I0524 15:49:32.544471 15114 sgd_solver.cpp:138] Iteration 40, lr = 0.001
I0524 15:49:40.604979 15114 solver.cpp:243] Iteration 50, loss = 0
I0524 15:49:40.605221 15114 solver.cpp:259]     Train net output #0: arm_loss = 0 (* 1 = 0 loss)
I0524 15:49:40.605296 15114 solver.cpp:259]     Train net output #1: odm_loss = 0 (* 1 = 0 loss)
I0524 15:49:41.358765 15114 sgd_solver.cpp:138] Iteration 50, lr = 0.001
I0524 15:49:48.693275 15114 solver.cpp:243] Iteration 60, loss = 0
I0524 15:49:48.693522 15114 solver.cpp:259]     Train net output #0: arm_loss = 0 (* 1 = 0 loss)
I0524 15:49:48.693544 15114 solver.cpp:259]     Train net output #1: odm_loss = 0 (* 1 = 0 loss)
I0524 15:49:48.693604 15114 sgd_solver.cpp:138] Iteration 60, lr = 0.001
I0524 15:49:56.767287 15114 solver.cpp:243] Iteration 70, loss = 0
I0524 15:49:56.767433 15114 solver.cpp:259]     Train net output #0: arm_loss = 0 (* 1 = 0 loss)
I0524 15:49:56.767449 15114 solver.cpp:259]     Train net output #1: odm_loss = 0 (* 1 = 0 loss)
I0524 15:49:57.122601 15114 sgd_solver.cpp:138] Iteration 70, lr = 0.001
I0524 15:50:04.454131 15114 solver.cpp:243] Iteration 80, loss = 0
I0524 15:50:04.454241 15114 solver.cpp:259]     Train net output #0: arm_loss = 0 (* 1 = 0 loss)
I0524 15:50:04.454253 15114 solver.cpp:259]     Train net output #1: odm_loss = 0 (* 1 = 0 loss)
I0524 15:50:04.777046 15114 sgd_solver.cpp:138] Iteration 80, lr = 0.001
I0524 15:50:12.264328 15114 solver.cpp:243] Iteration 90, loss = 0
I0524 15:50:12.264448 15114 solver.cpp:259]     Train net output #0: arm_loss = 0 (* 1 = 0 loss)
I0524 15:50:12.264464 15114 solver.cpp:259]     Train net output #1: odm_loss = 0 (* 1 = 0 loss)
I0524 15:50:12.587386 15114 sgd_solver.cpp:138] Iteration 90, lr = 0.001
I0524 15:50:20.822679 15114 solver.cpp:243] Iteration 100, loss = 0
I0524 15:50:20.824220 15114 solver.cpp:259]     Train net output #0: arm_loss = 0 (* 1 = 0 loss)
I0524 15:50:20.824316 15114 solver.cpp:259]     Train net output #1: odm_loss = 0 (* 1 = 0 loss)
I0524 15:50:21.173053 15114 sgd_solver.cpp:138] Iteration 100, lr = 0.001
I0524 15:50:29.037555 15114 solver.cpp:243] Iteration 110, loss = 0
I0524 15:50:29.037648 15114 solver.cpp:259]     Train net output #0: arm_loss = 0 (* 1 = 0 loss)
I0524 15:50:29.037660 15114 solver.cpp:259]     Train net output #1: odm_loss = 0 (* 1 = 0 loss)
I0524 15:50:29.364135 15114 sgd_solver.cpp:138] Iteration 110, lr = 0.001
I0524 15:50:36.191009 15114 solver.cpp:243] Iteration 120, loss = 0
I0524 15:50:36.191296 15114 solver.cpp:259]     Train net output #0: arm_loss = 0 (* 1 = 0 loss)
I0524 15:50:36.191318 15114 solver.cpp:259]     Train net output #1: odm_loss = 0 (* 1 = 0 loss)
I0524 15:50:36.191390 15114 sgd_solver.cpp:138] Iteration 120, lr = 0.001
I0524 15:50:45.883123 15114 solver.cpp:243] Iteration 130, loss = 0
I0524 15:50:45.883232 15114 solver.cpp:259]     Train net output #0: arm_loss = 0 (* 1 = 0 loss)
I0524 15:50:45.883252 15114 solver.cpp:259]     Train net output #1: odm_loss = 0 (* 1 = 0 loss)
I0524 15:50:46.242628 15114 sgd_solver.cpp:138] Iteration 130, lr = 0.001
I0524 15:50:55.017300 15114 solver.cpp:243] Iteration 140, loss = 0
I0524 15:50:55.023331 15114 solver.cpp:259]     Train net output #0: arm_loss = 0 (* 1 = 0 loss)
I0524 15:50:55.023352 15114 solver.cpp:259]     Train net output #1: odm_loss = 0 (* 1 = 0 loss)
I0524 15:50:55.023430 15114 sgd_solver.cpp:138] Iteration 140, lr = 0.001
I0524 15:51:03.233762 15114 solver.cpp:243] Iteration 150, loss = 0
I0524 15:51:03.234093 15114 solver.cpp:259]     Train net output #0: arm_loss = 0 (* 1 = 0 loss)
I0524 15:51:03.234143 15114 solver.cpp:259]     Train net output #1: odm_loss = 0 (* 1 = 0 loss)
I0524 15:51:03.597946 15114 sgd_solver.cpp:138] Iteration 150, lr = 0.001
I0524 15:51:11.431211 15114 solver.cpp:243] Iteration 160, loss = 0
I0524 15:51:11.431279 15114 solver.cpp:259]     Train net output #0: arm_loss = 0 (* 1 = 0 loss)
I0524 15:51:11.431293 15114 solver.cpp:259]     Train net output #1: odm_loss = 0 (* 1 = 0 loss)
I0524 15:51:11.431336 15114 sgd_solver.cpp:138] Iteration 160, lr = 0.001
I0524 15:51:18.636819 15114 solver.cpp:243] Iteration 170, loss = 0
I0524 15:51:18.636931 15114 solver.cpp:259]     Train net output #0: arm_loss = 0 (* 1 = 0 loss)
I0524 15:51:18.636950 15114 solver.cpp:259]     Train net output #1: odm_loss = 0 (* 1 = 0 loss)
I0524 15:51:18.988298 15114 sgd_solver.cpp:138] Iteration 170, lr = 0.001
I0524 15:51:26.973901 15114 solver.cpp:243] Iteration 180, loss = 0
I0524 15:51:26.974246 15114 solver.cpp:259]     Train net output #0: arm_loss = 0 (* 1 = 0 loss)
I0524 15:51:26.974263 15114 solver.cpp:259]     Train net output #1: odm_loss = 0 (* 1 = 0 loss)
I0524 15:51:27.676578 15114 sgd_solver.cpp:138] Iteration 180, lr = 0.001
I0524 15:51:34.939570 15114 solver.cpp:243] Iteration 190, loss = 0
I0524 15:51:34.939693 15114 solver.cpp:259]     Train net output #0: arm_loss = 0 (* 1 = 0 loss)
I0524 15:51:34.939707 15114 solver.cpp:259]     Train net output #1: odm_loss = 0 (* 1 = 0 loss)
I0524 15:51:35.725406 15114 sgd_solver.cpp:138] Iteration 190, lr = 0.001
I0524 15:51:43.889343 15114 solver.cpp:243] Iteration 200, loss = 0
I0524 15:51:43.889561 15114 solver.cpp:259]     Train net output #0: arm_loss = 0 (* 1 = 0 loss)
I0524 15:51:43.889582 15114 solver.cpp:259]     Train net output #1: odm_loss = 0 (* 1 = 0 loss)
I0524 15:51:43.889652 15114 sgd_solver.cpp:138] Iteration 200, lr = 0.001
I0524 15:51:51.997437 15114 solver.cpp:243] Iteration 210, loss = 0
I0524 15:51:51.997553 15114 solver.cpp:259]     Train net output #0: arm_loss = 0 (* 1 = 0 loss)
I0524 15:51:51.997571 15114 solver.cpp:259]     Train net output #1: odm_loss = 0 (* 1 = 0 loss)
I0524 15:51:52.360067 15114 sgd_solver.cpp:138] Iteration 210, lr = 0.001
I0524 15:52:01.192824 15114 solver.cpp:243] Iteration 220, loss = 0
I0524 15:52:01.193169 15114 solver.cpp:259]     Train net output #0: arm_loss = 0 (* 1 = 0 loss)
I0524 15:52:01.193189 15114 solver.cpp:259]     Train net output #1: odm_loss = 0 (* 1 = 0 loss)

**Comment by sfzhang15** [comment]
@Angzz 
Please check the generated IMDB whether the bbox value in it is right. Sometimes it may be all 0.

**Comment by zhangxixi0904** [comment]
if your bbox value is float type? I change the my annotation bbox value type from float to int and it works. @Angzz

**Comment by zhangxixi0904** [comment]
Hi! I see the the parameter resize_width and resize_height in create_data.sh is zero. What does it mean? Should I set it to a specific value? @sfzhang15

**Comment by Angzz** [comment]
@zhangxixi0904 Thanks very much, I changed the annos to int and it works!  I am also confused why the resize_width and resize_height are zero, I think it should be the same with the input image size. 

@sfzhang15 Thanks the author too, and I also want to know if I  can use bigger input size, beacause the objects in my dataset is too small all, Thanks again!!!

**Comment by sfzhang15** [comment]
@zhangxixi0904 @Angzz Hi,
1. The code of SSD to generate LMDB at [here](https://github.com/sfzhang15/RefineDet/blob/master/src/caffe/util/io.cpp#L321-L324) has some problem and we forgot to fix it. Now we have fixed it to support the float type of bbox. 
2. The parameter resize_width and resize_height in create_data.sh is zero that means do not resize the image, see [here](https://github.com/sfzhang15/RefineDet/blob/master/src/caffe/util/io.cpp#L113). So you should not set it to a specific value.
3. @Angzz You can use bigger input size to detect small objects if the GPU memory allows.

**Comment by Angzz** [comment]
@sfzhang15 Thanks a lot, I will have a try and tell you the latest progress.
